test/hotspot/jtreg/runtime/linkResolver/InterfaceObjectTest.java

```diff
@@ -42,19 +42,19 @@ public Object clone() {
         System.out.println("In InterfaceObjectTest's clone() method\n");
         return null;
     }
-
-    public void finalize() throws Throwable {
+    public String toString() {
         try {
-            System.out.println("In InterfaceObjectTest's finalize() method\n");
+            System.out.println("In InterfaceObjectTest's toString() method\n");
         } catch (Throwable t) {
             throw new AssertionError(t);
         }
+        return "InterfaceObjectTest";
     }
 
     public static void tryIt(ICloneExtend o1) {
         try {
             Object o2 = o1.clone();
-            o1.finalize();
+            o1.toString();
         } catch (Throwable t) {
             throw new AssertionError(t);
         }
```

Comment:
I don't think you need toString() since it's public and will be found through the I interface class.  I don't know what the purpose of tryIt is. If you change the call in tryIt to clone, I think that will show that you can find clone() only if it's overridden in the class.

Reply:
Yes. As discussed above only protected methods should be part of this test, so no need to add toString().